hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -93,6 +93,11 @@ def __init__(
 
         self.target_selector = TargetSelector(self.random)
 
+        # Previously-tested byte streams are recorded in a prefix tree, so that
+        # we can avoid testing the same stream twice (in some cases),
+        # and so that we can proactively generate stream prefixes that haven't
+        # been tried before.
```

Comment:
This misses a use case in that it's not just streams that we have previously seen before we avoid testing, it's also streams which we know don't have enough data - if we know that a stream is a strict prefix of some previous stream we've seen, then what will happen is that it will make all the same decisions right up until the point where it reads off the end.